cpp/src/io/csv/csv_gpu.cu

```diff
@@ -554,9 +554,9 @@ __global__ void __launch_bounds__(csvparse_block_dim)
                       device_span<char const> const data,
                       device_span<column_parse::flags const> const column_flags,
                       device_span<uint64_t const> const row_offsets,
-                      cudf::data_type *dtypes,
-                      void **columns,
-                      cudf::bitmask_type **valids)
+                      device_span<cudf::data_type const> const dtypes,
+                      device_span<void *> const columns,
+                      device_span<cudf::bitmask_type *> const valids)
```

Comment:
Here I dislike the const a bit more, could be confusing for folks who are not that familiar with spans.
```suggestion
                      device_span<void *> columns,
                      device_span<cudf::bitmask_type *> valids)
```

Reply:
I left the `const` intentionally so it's known those variables never change. One less thing to think about.